metron-platform/metron-parsers/src/main/java/org/apache/metron/parsers/topology/ParserTopologyBuilder.java

```diff
@@ -91,14 +101,14 @@ public Config getTopologyConfig() {
    */
   public static ParserTopology build(String zookeeperUrl,
                                       Optional<String> brokerUrl,
-                                      String sensorType,
-                                      ValueSupplier<Integer> spoutParallelismSupplier,
-                                      ValueSupplier<Integer> spoutNumTasksSupplier,
+                                      List<String> sensorTypes,
+                                      ValueSupplier<List> spoutParallelismSupplier,
```

Comment:
This is a nit, but is there any reason why we couldn't make these `ValueSupplier<List<Integer>>` and `ValueSupplier<List<Map>>`?

Reply:
I actually forgot to ask about this in the main thing.  It's because of the use of the Class argument in ValueSupplier breaks the typing involved elsewhere. It's the same reason it's ValueSupplier<Map> instead of ValueSupplier<Map<Foo, Bar>> elsewhere.

Having said that, it seems like it's entirely unnecessary to have the class argument at all and we could just actually type everything everywhere and just remove the argument entirely. I tried this out briefly and it seemed fine, but I didn't spin it up through full dev since it seemed optional to do.

Reply:
Ah, I see.  man, I miss C++ templates sometimes.  I retract the nit; don't worry about it.